### PR TITLE
Fix and update pyannote CLI

### PR DESCRIPTION
**Pull request by clement-pages**
The purpose of this pull request is to update `pyannote.audio` cli by fixing the following points:
- update the way the task is setup, taking into account data preparation with `model.prepare_data()` for the `pyannote-audio-train` command
- takes into account the fact that checkpointing is not available when the `trainer` is in `fast_dev_run` mode.
- added `registry` for the `pyannote-audio-eval` command

This pull request also adds a few unit tests for these two commands

### CODE REVIEW

File: pyannote/audio/cli/evaluate_config/config.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 model: ???
 protocol: ???
+registry: ???
```
Comment (hbredin) [inline pyannote/audio/cli/evaluate_config/config.yaml:3]: Can we still make it optional?
Indeed, `pyannote.database` will automatically load `database.yml` in current directory.

File: pyannote/audio/cli/train.py [hunk 2]
```diff
@@ -77,7 +77,8 @@ def train(cfg: DictConfig) -> Optional[float]:
     fine_tuning = cfg.model["_target_"] == "pyannote.audio.cli.pretrained"
     model = instantiate(cfg.model)
     model.task = task
```
Comment (hbredin) [inline pyannote/audio/cli/train.py:79]: In some code I am writing in a private repo, I need to access `task` as soon as `Model.__init__`. So I recommended doing this:

```suggestion
    model = instantiate(cfg.model, task=task)
```

File: tests/data/simple_model_on_speaker_diarization_task.ckpt
Comment (hbredin) [inline tests/data/simple_model_on_speaker_diarization_task.ckpt:1]: I'd rather avoid adding such a model in the repo.
Would it possible to use https://huggingface.co/pyannote/TestModelForContinuousIntegration for instance? 

If this model needs to be updated to a newer version, we can do that too.

Comment (clement-pages) [reply tests/data/simple_model_on_speaker_diarization_task.ckpt:1]: > Would it possible to use https://huggingface.co/pyannote/TestModelForContinuousIntegration for instance?

I'm going to test with this model.

Comment (clement-pages) [reply tests/data/simple_model_on_speaker_diarization_task.ckpt:1]: Indeed, I think the model need to be updated. I have the following error when testing:
```shell
Error executing job with overrides: ['model=pyannote/TestModelForContinuousIntegration', '+registry=./tests/data/database.yml', 'protocol=Debug.SpeakerDiarization.Debug']
Traceback (most recent call last):
  File "/home/clement-pages/micromamba/envs/pyannote/bin/pyannote-audio-eval", line 33, in <module>
    sys.exit(load_entry_point('pyannote.audio', 'console_scripts', 'pyannote-audio-eval')())
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/main.py", line 94, in decorated_main
    _run_hydra(
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/_internal/utils.py", line 394, in _run_hydra
    _run_app(
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/_internal/utils.py", line 457, in _run_app
    run_and_report(
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/_internal/utils.py", line 223, in run_and_report
    raise ex
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/_internal/utils.py", line 220, in run_and_report
    return func()
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/_internal/utils.py", line 458, in <lambda>
    lambda: hydra.run(
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/_internal/hydra.py", line 132, in run
    _ = ret.return_value
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/core/utils.py", line 260, in return_value
    raise self._return_value
  File "/home/clement-pages/micromamba/envs/pyannote/lib/python3.10/site-packages/hydra/core/utils.py", line 186, in run_job
    ret.return_value = task_function(task_cfg)
  File "/home/clement-pages/pyannote-audio/pyannote/audio/cli/evaluate.py", line 80, in evaluate
    _ = metric(reference, hypothesis(file), uem=uem)
  File "/home/clement-pages/pyannote-audio/pyannote/audio/cli/evaluate.py", line 71, in hypothesis
    return Inference.trim(
  File "/home/clement-pages/pyannote-audio/pyannote/audio/core/inference.py", line 670, in trim
    scores.data.ndim == 3
AssertionError: Inference.trim expects (num_chunks, num_frames, num_classes)-shaped `scores`
```

Comment (clement-pages) [reply tests/data/simple_model_on_speaker_diarization_task.ckpt:1]: I have trained a new version of this model, but I still have the same error, so this is not due to a version problem

Comment (hbredin) [reply tests/data/simple_model_on_speaker_diarization_task.ckpt:1]: I think the model needs to be a segmentation model, not a VAD model.

Comment (hbredin) [reply tests/data/simple_model_on_speaker_diarization_task.ckpt:1]: I suggest you train this model for one epoch and we will share it on HF for CI.

```python
task = SpeakerDiarization(ami, duration=5.0, max_speakers_per_chunk=3, max_speakers_per_frame=2, ...)
model = SimpleSegmentationModel(task=task)
```

**Pull request merged**